tools/online_editor/src/editor_widget.ts

```diff
@@ -485,19 +487,23 @@ class EditorPaneWidget extends Widget {
         try {
             const response = await fetch(uri.toString());
             if (!response.ok) {
-                alert(
-                    "Failed to download data from " +
-                        uri +
-                        ":\n" +
-                        response.status +
-                        " " +
-                        response.statusText,
-                );
+                if (raise_alert) {
+                    alert(
```

Comment:
I think that alert() provides a very bad user experience and we should not use it at all. I'd be more than happy to expand on the "why" if it's not obvious (please tell me). Meanwhile, if absolutely necessary, please use the HTML dialog element.

Reply:
I'd argue the logic might be worth changing as well. In a desktop app you'd probably show a "Wizard": 

- First page: "Enter the URL"
- Second page: "Importing files, please wait" (progress bar)
- Third page: Import complete, X files imported, a note about that it can be exported into a new gist, etc.

If something goes wrong during the import, then the second wizard page is exactly where we can show a list of what happened to the user. From "The url doesn't work at all" to "It all worked, but this one file was missing". Then as a user I can choose to continue anyway (and have a possibly incomplete set of files loaded for editing), I can choose "Back" to correct the URL and try again, or I can choose "Cancel" and the documents I had previously opened are still there.

But the current workflow is: Paste a url, pray that it works, if I get an alert() then all I can do is click it away, but I'm now stuck in a pretty useless intermediate state in the editor.